app/src/main/java/com/sun_asterisk/foodies/data/source/repository/RecipesRelatedRepository.kt

```diff
@@ -0,0 +1,21 @@
+package com.sun_asterisk.foodies.data.source.repository
+
+import com.sun_asterisk.foodies.data.model.RecipesRelated
+import com.sun_asterisk.foodies.data.source.datasource.RecipesRelatedDataSource
+import com.sun_asterisk.foodies.data.source.remote.OnFetchDataJsonListener
+import com.sun_asterisk.foodies.data.source.remote.RecipesRelatedRemoteDataSource
+
+class RecipesRelatedRepository private constructor(private val remote: RecipesRelatedDataSource.Remote) {
+
+    private object Holder {
+        val INSTANCE = RecipesRelatedRepository(remote = RecipesRelatedRemoteDataSource.instance)
+    }
+
+    fun getRecipesRelatedInfo(listener: OnFetchDataJsonListener<MutableList<RecipesRelated>>, ingredientName: String?) {
+        remote.getRecipesRelatedInfo(listener, ingredientName)
+    }
+
+    companion object {
+        val instance: RecipesRelatedRepository by lazy { Holder.INSTANCE }
```

Comment:
```suggestion
        private var instance: RecipesRelatedRepository? = null
        fun getInstance(remote: RecipesRelatedDataSource.Remote) =
            instance ?: RecipesRelatedRepository(remote).also { instance = it }
```
Em không nên khởi tạo ngay bên trong class mà nên truyền nó vào từ bên ngoài để dễ sàng cho việc test sau này